akka-actor/src/main/scala/akka/util/ConfigValidation.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.util
+
+import scala.util.Try
+import scala.util.control.NonFatal
+
+import akka.annotation.InternalApi
+import com.typesafe.config.Config
+
+@InternalApi
+private[akka] object ConfigValidation {
+
+  /** Allow usage outside validation. */
+  def isValid(path: String, config: Config): Boolean =
+    path != null && path.nonEmpty && config.hasPath(path)
+
+  def message(key: String): String =
+    s"Configuration [$key] was expected but is not configured."
+
+  trait ConfigVal[A] {
+    def value(path: String, config: Config): A
+  }
+
+  implicit def stringConfigVal: ConfigVal[String] = new ConfigVal[String] {
+    def value(path: String, config: Config): String = config.getString(path)
+  }
+  implicit def booleanConfigVal: ConfigVal[Boolean] = new ConfigVal[Boolean] {
+    def value(path: String, config: Config): Boolean = config.getBoolean(path)
+  }
+
+  final implicit class ConfigValidationOps(val config: Config) extends AnyVal {
+
+    /** Fail fast on config error. */
+    def as[A](path: String)(implicit reader: ConfigVal[A]): A =
+      if (isValid(path, config)) {
```

Comment:
we almost always have config properties in reference.conf and that should be encouraged because it's **the** place to document existence and meaning of a config property, so using `hasPath` is an anti-pattern

Reply:
I got `hasPath` from yours and the existing code though so I thought it was a fair example on what boilerplate we have.

Reply:
There are cases where it's valid usage because it's an open ended config section where we can't have it in reference.conf. Those usages are often associated with more logic than just getting a value or using a default.